projects/optic/src/commands/oas/captures/streams/sources/proxy.ts

```diff
@@ -39,7 +39,7 @@ export class ProxyInteractions {
       targetCA?: Array<{ cert: Buffer | string }>;
     } = {}
   ): Promise<[ProxyInteractions, string, string]> {
-    let { host, protocol } = new URL(targetHost);
+    let { host, protocol, origin: targetOrigin } = new URL(targetHost);
     if (targetHost.includes('/')) {
       // accept urls to be passed in rather than pure hosts
       targetHost = host;
```

Comment:
This function is a little overloaded, but we're generating the transparent proxy and the mockttp server here. I'm fixing specifically the mockttp server case, and leaving the transparent proxy host logic intact